src/database/mod.rs

```diff
@@ -243,7 +243,7 @@ impl DatabaseConnection for SqliteConn {
         self.db_address(address).map(|db_addr| {
             (
                 db_addr.derivation_index,
-                address == &db_addr.change_address.assume_checked(),
+                address.to_string() == db_addr.change_address.assume_checked().to_string(),
```

Comment:
Can you add a small comment with the rationale please? It's sneaky and i don't want us to forget about it.

Reply:
I've now added a comment here.